receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -351,9 +351,14 @@ func (r *pReceiver) getScrapeConfigsResponse(baseURL string) (map[string]*config
 }
 
 func (r *pReceiver) applyCfg(cfg *PromConfig) error {
+	scrapeConfigs, err := (*config.Config)(cfg).GetScrapeConfigs()
+	if err != nil {
+		return fmt.Errorf("could not get scrape configs: %w", err)
+	}
+
 	if !enableNativeHistogramsGate.IsEnabled() {
 		// Enforce scraping classic histograms to avoid dropping them.
-		for _, scrapeConfig := range cfg.ScrapeConfigs {
+		for _, scrapeConfig := range scrapeConfigs {
```

Comment:
Here we have the limitation that the `scrapeConfig.ScrapeClassigHistograms` property is only applied to the scrapeconfigs provided directly in the config, but not the ones via `scrape_config_files`, as the `scrapeManager` internally calls `GetScrapeConfigs()`, which leads to the scrapeConfigs being read from the file system again.
One potential solution could be to set `cfg.ScrapeConfigs` here to the result of `GetScrapeConfigs()`, and clear the `ScrapeConfigFiles` property, but this feels a bit hacky. What do you think @dashpole ?